main.go

```diff
@@ -92,7 +92,15 @@ func main() {
 	})
 
 	svr.AddFunc(func(ctx context.Context) {
-		srv := server.New(ctx, ds, mq, rnr, tasks, server.DefaultEnqueue)
+		srv := server.New(
+			ctx,
+			ds,
+			mq,
+			rnr,
+			tasks,
+			server.DefaultEnqueue,
+			server.EnableShutdownEndpoint(halt),
```

Comment:
In the end, this will be addressed by https://github.com/iron-io/functions/issues/462